docs/topics/Automated-Checks.md

```diff
@@ -0,0 +1,259 @@
+4. Do not include entities which are not `Verified` in the tournament rating algorithm.
+    * This provides an added benefit of ensuring all generated statistics are valid. Even with manually submitted data, humans make mistakes. If unverified data is introduced into the rating & statistics systems, users will notice invalid statistics and the rating ladder itself will not be completely accurate.
+
+## Entities
+
+The following entities are part of this processing pipeline:
+
+* `Tournaments`
+* `Matches`
+* `Games`
+* `GameScores`
+
+## Statuses
+
+Each entity has `VerificationStatus`, `ProcessingStatus`, and `RejectionReason` fields. These fields are referenced and changed by the DataWorkerService as they move through the processing flow.
+
+### `VerificationStatus`
+
+Each entity shares the same `VerificationStatus` type. This type contains the following statuses:
+
+* `None`: The entity has yet to be processed automatically.
+* `PreRejected`: Based on the system's rules, this entity should be rejected.
+* `PreVerified`: The system did not find anything wrong, awaiting human review.
+* `Rejected`: A human marked this entity as rejected.
+* `Verified`: A human marked this entity as verified.
+
+### `ProcessingStatus`
+
+Each entity has a unique `ProcessingStatus` type associated with it. This flag is self-explanatory: it indicates how far along an entity is in the processing pipeline.
+
+For example, consider `TournamentProcessingStatus`:
+
+1. `NeedsApproval`: The tournament is submitted but waiting approval from a verifier.
+2. `NeedsMatchData`: Match data needs to be fetched via the osu! API.
+3. `NeedsAutomationChecks`: The tournament, and all of its children, are awaiting automation checks.
+4. `NeedsVerification`: Awaiting human review
+5. `NeedsStatCalculation`: After human review, process statistics (must be complete before it is eligible for inclusion in the rating system).
+6. `Done`: Processing is completed. `Verified` tournaments with this status are eligible for inclusion in the rating system.
+
+### `RejectionReason`
+
+Each entity has a custom `RejectionReason` type with various flags which may cause it to be marked as `PreRejected`. Flags can be combined with each other to form a set of reasons. For example, a `Game` could be marked as `PreRejected` by the system due to `NoScores` and `BeatmapNotPooled`.
+
+## Flow
+
+Automation checks are performed in the following order.
+
+```Mermaid
+flowchart LR;
+    GameScore --> Game --> Match --> Tournament
+```
+
+> This allows the parent entities to have the context of how their children faired during the automated checks process.
+> 
+
+### Tournament
+
+```Mermaid
+flowchart TD;
+   A[Is the count of PreVerified and/or Verified matches >= 0?]
```

Comment:
```suggestion
   A[Is the count of PreVerified and/or Verified matches > 0?]
```
(I guess also double check whether the 80% below is inclusive or not)

Reply:
Seems to be inclusive